ccm_web/server/src/config.ts

```diff
@@ -44,60 +44,83 @@ export interface Config {
 const logger = baseLogger.child({ filePath: __filename })
 
 const isString = (v: unknown): v is string => typeof v === 'string'
-const isNumber = (v: unknown): v is number => typeof v === 'number' && !isNaN(v)
+const isNotEmpty = (v: string): boolean => v.length > 0
+
+const isNumber = (v: unknown): v is number => typeof v === 'number'
+const isNotNan = (v: number): boolean => !isNaN(v)
+
+// Tests if an unknown value is one of four allowed string log levels
 const isLogLevel = (v: unknown): v is LogLevel => {
   return isString(v) && ['debug', 'info', 'warn', 'error'].includes(v)
 }
 
+// Handles some edge cases and casts all other values using Number
+const prepNumber = (value: string | undefined): string | number | undefined => {
+  return (value === undefined) ? undefined : value.trim() === '' ? value : Number(value)
+}
```

Comment:
I did some last minute testing of these config changes, and I found an edge case I wasn't properly handling. There's an oddity with `Number()` that empty or all-whitespace strings are converted to `0`, which didn't seem like desirable behavior. Now I'm just returning those values so that they don't pass the type check. This number handling is a pain 😢